extension-manifest-v3/src/pages/settings/components/custom-filters.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+
+import { html, store } from 'hybrids';
+import { detectFilterType } from '@cliqz/adblocker';
+
+import * as converter from '/utils/dnr-converter.js';
+import CustomFiltersInput from '../store/custom-filters-input.js';
+import { asyncAction } from './devtools.js';
+
+function parseFilters(text = '') {
+  return text
+    .split('\n')
+    .map((f) => f.trim())
+    .filter(Boolean)
+    .reduce(
+      (filters, filter) => {
+        const filterType = detectFilterType(filter);
+        if (filterType === 1) {
+          // NETWORK
+          filters.networkFilters.add(filter);
+        } else if (filterType === 2) {
+          // COSMETIC
+          filters.cosmeticFilters.add(filter);
+        } else {
+          filters.errors.push(`Filter not supported: '${filter}'`);
+        }
+        return filters;
+      },
+      {
+        networkFilters: new Set(),
+        cosmeticFilters: new Set(),
+        errors: [],
+      },
+    );
+}
+
+async function submitFilters(host) {
+  const { networkFilters, cosmeticFilters } = host.filters;
+
+  // Update DNR
+  if (__PLATFORM__ !== 'firefox') {
+    const dnrRules = [];
+    const dnrErrors = [];
+    const results = await Promise.allSettled(
+      [...networkFilters].map((filter) => converter.convert(filter)),
```

Comment:
no need for array spread

Reply:
It's a Set. Look above. I've changed that to simply remove duplicated filters (I discovered them while testing).